components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -161,15 +153,35 @@ class Nimbus : NimbusApi {
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        try {
+            // We need to launch this off of the main thread because it involves a network request
+            // within the nimbus-sdk.
+            scope.launch {
+                nimbus.updateExperiments()
+                experimentsUpdated()
+            }
+        } catch (e: ErrorException.RequestError) {
+            logger.info("Error fetching experiments from endpoint: $e")
+        } catch (e: ErrorException.InvalidExperimentResponse) {
+            logger.info("Invalid experiment response: $e")
+        }
     }
 
     override fun optOut(experimentId: String) {
         if (!isInitialized) return
         nimbus.optOut(experimentId)
     }
 
+    // When experiments are updated or when the library is initialized, we need to do some tasks
+    // like recording telemetry and invoking the callback.
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun experimentsUpdated() {
```

Comment:
```suggestion
    // like recording telemetry and invoking the callback.
    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
    private fun experimentsUpdated() {
```

I don't know what the visibility is at this point. If we're unit testing (e.g. in `NimbusTest`) then they're visible, but otherwise, are they `internal` or `private`.